index.bs

```diff
@@ -266,34 +266,23 @@ are fairly clearly sniffable by "examining the structure of other headers and by
 availability and semantics of the features introduced or modified between releases of a particular
 browser" [[Janc2014]]).
 
-To <dfn abstract-op local-lt="set-ua">return the `Sec-CH-UA` value for a request</dfn>, given a
-<a href="https://wicg.github.io/client-hints-infrastructure/#environment-settings-object-client-hints-set">client hints set</a> (|set|),
-[=user agents=] MUST:
+To <dfn abstract-op local-lt="set-ua">return the `Sec-CH-UA` value for a request</dfn>, [=user agents=] MUST:
 
-1.  Let |value| be a [=Structured Header=] object whose value is a [=structured header/list=].
+1.  Let |list| be an [=/list=], initially empty.
 
-2.  Let |version| be the [=user agent=]'s [=user agent/full version=] if |set|
-    [=list/contains=] `UA`, and the [=user agent=]'s [=user agent/significant version=] otherwise.
+2. For each {{NavigatorUABrandVersion}} in [=user agent/UA list=]:
 
-3.  Let |ua| be a string whose value is the [=string/concatenation=] of the [=user agent=]'s
-    [=user agent/brand=], the string ";v=", and |version|.
+    1. Let |parameter| be a [=dictionary=], initially empty.
 
-4.  [=list/Append=] |ua| to |value|.
+    2. Set |parameter|["param_name"] to "v".
 
-5.  The [=user agent=] MAY execute the following steps:
+    3. Set |parameter|["param_value"] to {{NavigatorUABrandVersion/version}}.
 
-    1.  [=list/Append=] additional items to |value| containing arbitrary brand and version
-        combinations.
+    2. Let |pair| be a tuple comprised of {{NavigatorUABrandVersion/brand}} and |parameter|.
 
-    2.  Randomize the order of the items in |value|.
+    3. Append |pair| to |list|.
```

Comment:
FWIW you could use literal syntax to shorten these three steps into

> Append (_brandVersion_'s `brand`, «[ "`param_name`" → "`v`", "`param_value`" → _brandVersion_'s `version` ]»).

Not sure if that's clearer, but just wanted to offer it as an option.

(I guess that assumes that structured header's "ordered Dictionary" concept is the same as Infra's "ordered map", but that seems safe.)

Reply:
That's good to know. I'll keep it as is, as the current version is clearer IMO